Lifetimed indexing for array accessors (#44)
toku-sa-n/accessor: README.md

![Crates.io](https://img.shields.io/crates/l/accessor)
[![docs.rs](https://docs.rs/accessor/badge.svg)](https://docs.rs/accessor)

Accessors to read and write physical memory volatilely, such as performing memory-mapped I/O.

This crate provides accessors to the value at a specific physical memory address.

> The accessed type doesn't have to implement [`Copy`], but be aware that reading and writing the value creates a bitwise copy it.

Reviewer: ```suggestion
The accessed type doesn't have to implement [`Copy`], but be aware that reading and writing the value creates a bitwise copy of it.
```

Accessors are similar to pointers with volatile read/writes
(or for those who are familiar with crate [volatile(~v0.3.0)](https://docs.rs/volatile/0.3.0/volatile/index.html), pointers of volatile wrappers)
but also designed to refer correct physical addresses even in virtual memory mode,
once an appropriate physical-to-virtual memory mapper is specified.

When an accessor is created, the physical memory is mapped into virtual memory, with the help of the
mapper implemented by the crate user. The methods of accessors allow access to the value at the
specified physical address. Once an accessor is dropped, the mapped memory is unmapped.

If one has full control of physical memory addresses(e.g. developing their own kernel),

> a 'virtual' address is equal to the physical one, and the mapper should map any address into itself.

Reviewer: ```suggestion
a 'virtual' address may be equal to the physical one, then the mapper should map any address into itself.
```
